src/compiler/program.ts

```diff
@@ -626,13 +627,30 @@ namespace ts {
                 onlyRecordFailures = !directoryProbablyExists(directory, state.host);
             }
         }
-        return forEach(extensions, tryLoad);

```

Comment:
consider this pattern, split the function into a loadModuleFromFile and loadModuleFromFileWorker. the worker is never called directly except from loaModulefronFile. and loadModuleFromFile becomes:
```ts
    function loadModuleFromFile(candidate: string, extensions: string[], failedLookupLocation: string[], onlyRecordFailures: boolean, state: ModuleResolutionState): string {
        return loadModuleFromFileWorker(candidate, extensions, failedLookupLocation, onlyRecordFailures, state) ||
            hasJavaScriptFileExtension(candidate) ? loadModuleFromFileWorker(removeFileExtension(candidate), extensionsm failedLookupLocation, onlyRecordFailures, state) : undefined;
    }
```